### DISCUSSION TITLE
how to set baseUrl for useFetch in Nuxt 3

### DISCUSSION THREAD

**Discussion by lecneri** [nuxt/nuxt#16805] [Questions]
Hello,

My API server is different from my nuxt server.
So instead of every component I make a full request I'd like to set baseUrl for every request.

Here's what I've done so far. 
1st Attempt 
Try to use composables, but the result is not what I'd expected. I've read that nuxt 3 uses ohmyfetch so I did this on composables/usefetch
```
import { $fetch } from 'ohmyfetch'
export default $fetch.create({
  baseURL: 'http://myotherserver:8000'
});
```
It works but it does not return AsyncData, which makes more annoying to get pending, error

2nd Attempt
make an interceptor server/api ou server/middleware, but it doesn't feels right and troublesome to useBody, useQuery to recreate and forward request

3rd Attempt
try to modify nuxt.config.ts and modify modules to use baseURL for ohmyfetch, but I didn't figure out if is this right or how to make it work.

So I'd like know what's the best way to set baseURL.

Thanks in advance

**Comment by sevillaarvin** [comment]
> Here's what I've done so far. 1st Attempt Try to use composables, but the result is not what I'd expected. I've read that nuxt 3 uses ohmyfetch so I did this on composables/usefetch
> 
> ```
> import { $fetch } from 'ohmyfetch'
> export default $fetch.create({
>   baseURL: 'http://myotherserver:8000'
> });
> ```

Based on fetch.ts https://github.com/nuxt/framework/blob/2894a75bc8cd6430a5945f411a8e79dac67288a5/packages/nuxt/src/app/composables/fetch.ts#L90, I think you can just wrap it with `useAsyncData`:

```js
useAsyncData(() => mycomposablefetch())
```

**Reply by lecneri** [reply]
I'll try this out. Thank you for your reply
